internal/pkg/template/templates/workloads/services/static-site/cf.yml

```diff
@@ -3,6 +3,14 @@
 AWSTemplateFormatVersion: 2010-09-09
 Description: CloudFormation template that represents a static site backed by S3.
 
+{{- /*
+!¡!¡!¡!¡!¡!¡
+If you are adding resources to this template, make sure to add the resources to
```

Comment:
Is `TriggerStateMachineAction`  _always_ expected to be the last resource to get CREATE/UPDATEd? If so, I wonder if it's worth it to write a test that errors out if a resource is in the template but not in `TriggerStateMachineAction.DependsOn`  💭

Reply:
Agreed. We could probably do it after the release. I wish there's an integrated method doing that instead of always adding to the list...

Reply:
I'm not even sure that it needs to be added to the `DependsOn` thought 💭 

Like if we add a completely random resource that can indeed be created in parallel with the `TriggerStateMachine` custom resource, we wouldn't need to add it as a dependsOn *methinks*

Reply:
my main thought here is to prevent this flow, even though it's not quite perfect:
1. Resources start being created
2. Trigger state machine which copies files into s3
3. Some other resource fails, triggering rollback

At this state, some/all files were "deployed" to s3, and we don't have a mechanism to roll that back. By making the "start the copy" action depend on everything else, at least we will only do the copy if all the other resources have successfully been created/updated. It doesn't solve for failures _within_ the state machine/copy lambda, though.

Reply:
your reasoning makes sense to me 👍 

Maybe we can strike a good balance: the default is that the test ensures that the resource is added as a dependency, _however_ there should be a way to modify the test so that I can explicitly opt-out a dependsOn.

What do you think?

Reply:
yeah, i think that's a good balance! we'll have to find a way to generate all possible versions of the template to validate....or something but i think the test route is a good idea.